### PR TITLE
Expose storage media type as env variable

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #40636

@mml

### CODE REVIEW

File: cluster/gce/config-test.sh [hunk 1]
```diff
@@ -195,8 +193,10 @@ KUBE_UP_AUTOMATIC_CLEANUP=${KUBE_UP_AUTOMATIC_CLEANUP:-false}
 # is only supported in trusty or GCI.
 TEST_CLUSTER="${TEST_CLUSTER:-true}"
 
-# Storage backend. 'etcd2' supported, 'etcd3' experimental.
+# Storage backend. 'etcd2' and 'etcd3' are supported.
 STORAGE_BACKEND=${STORAGE_BACKEND:-}
+# Storage media type: application/json and application/vnd.kubernetes.protobuf are supported.
```
Comment (fgrzadkowski) [inline cluster/gce/config-test.sh]: Would it make sense to verify this flag somewhere?

File: cluster/kubemark/gce/config-default.sh [hunk 2]
```diff
@@ -73,7 +73,10 @@ KUBEMARK_MASTER_COMPONENTS_QPS_LIMITS="${KUBEMARK_MASTER_COMPONENTS_QPS_LIMITS:-
 # ContentType used by all components to communicate with apiserver.
 TEST_CLUSTER_API_CONTENT_TYPE="${TEST_CLUSTER_API_CONTENT_TYPE:-}"
 # ContentType used to store objects in underlying database.
-TEST_CLUSTER_STORAGE_CONTENT_TYPE="${TEST_CLUSTER_STORAGE_CONTENT_TYPE:-}"
+TEST_CLUSTER_STORAGE_CONTENT_TYPE=""
```
Comment (fgrzadkowski) [inline cluster/kubemark/gce/config-default.sh]: Would it make sense to modify this var name as well?

File: cluster/gce/config-test.sh [hunk 3]
```diff
@@ -130,11 +130,9 @@ TEST_CLUSTER_RESYNC_PERIOD="${TEST_CLUSTER_RESYNC_PERIOD:---min-resync-period=3m
 
 # ContentType used by all components to communicate with apiserver.
 TEST_CLUSTER_API_CONTENT_TYPE="${TEST_CLUSTER_API_CONTENT_TYPE:-}"
-# ContentType used to store objects in underlying database.
-TEST_CLUSTER_STORAGE_CONTENT_TYPE="${TEST_CLUSTER_STORAGE_CONTENT_TYPE:-}"
 
 KUBELET_TEST_ARGS="${KUBELET_TEST_ARGS:-} --max-pods=110 --serialize-image-pulls=false --outofdisk-transition-frequency=0 ${TEST_CLUSTER_API_CONTENT_TYPE}"
-APISERVER_TEST_ARGS="${APISERVER_TEST_ARGS:-} --runtime-config=extensions/v1beta1 ${TEST_CLUSTER_DELETE_COLLECTION_WORKERS} ${TEST_CLUSTER_MAX_REQUESTS_INFLIGHT} ${TEST_CLUSTER_STORAGE_CONTENT_TYPE}"
```
Comment (fgrzadkowski) [inline cluster/gce/config-test.sh]: Aren't we using `TEST_CLUSTER_STORAGE_CONTENT_TYPE` in some of our e2e tests? I don't see any files modified there.

Comment (wojtek-t) [inline cluster/gce/config-test.sh]: I just modified those tests to set both:
https://github.com/kubernetes/test-infra/pull/1921

Once this one is merged, I will clean the former one. 

But this should work now.

File: cluster/gce/config-test.sh (hunk 1 above)
Comment (wojtek-t) [inline cluster/gce/config-test.sh]: Hmm - I don't know what are the rules, but I didn't see any validation for any of our envs anywhere.
So I think it's not that necessary.

File: cluster/kubemark/gce/config-default.sh (hunk 2 above)
Comment (wojtek-t) [inline cluster/kubemark/gce/config-default.sh]: Fixed.

File: cluster/gce/config-test.sh (hunk 1 above)
Comment (fgrzadkowski) [inline cluster/gce/config-test.sh]: :(

**Review by fgrzadkowski — APPROVED** [review]

**Comment by wojtek-t** [comment]
@k8s-bot verify test this, issue: #IGNORE (timeout downloading gazel)

**Pull request merged**